File: Number Printer/ConsoleApp1/bin/Release/Brother bPAC3 SDK/Samples/VBNET/Badge/Form1.vb
'*******************************************************************
'
'       b-PAC 3.0 Component Sample (Badge)
'
'       (C)Copyright Brother Industries, Ltd. 2009
'
'*******************************************************************
Option Explicit On

Public Class Badge
    Const sPath = "C:\Program Files\Brother bPAC3 SDK\Templates\Badge.lbx"

    '********************************************************
    '   Open and Print a spcified file.
    '********************************************************
    Public Sub DoPrint()
        Dim objDoc As bpac.Document
        objDoc = CreateObject("bpac.Document")
        If (objDoc.Open(sPath) <> False) Then
            objDoc.GetObject("objName").Text = txtName.Text
            objDoc.GetObject("objCompany").Text = txtCompany.Text

            'objDoc.SetMediaByName(objDoc.Printer.GetMediaName, True)
            objDoc.StartPrint("", bpac.PrintOptionConstants.bpoDefault)
            objDoc.PrintOut(1, bpac.PrintOptionConstants.bpoDefault)
            objDoc.EndPrint()
            objDoc.Close()
        End If
    End Sub

    Private Sub btnPrint_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnPrint.Click
        DoPrint()
    End Sub

    Private Sub btnClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub
End Class

File: Number Printer/ConsoleApp1/bin/Release/Brother bPAC3 SDK/Samples/VCS/NamePlt/Form1.cs
/***********************************************************************
			b-PAC 3.0 Component Sample (TestPrint)

			(C)Copyright Brother Industries, Ltd. 2009
************************************************************************/

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using bpac;

namespace TestPrint
{
    public partial class Form1 : Form
    {
		private const string TEMPLATE_DIRECTORY = @"C:\Program Files\Brother bPAC3 SDK\Templates\";	// Template file path
		private const string TEMPLATE_SIMPLE = "NamePlate1.LBX";	// Template file name
		private const string TEMPLATE_FRAME = "NamePlate2.LBX";		// Template file name

        public Form1()
        {
            InitializeComponent();
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			this.cmbTemplate.SelectedIndex = 0;
		}

		private void btnPrint_Click(object sender, EventArgs e)
		{
			string templatePath = TEMPLATE_DIRECTORY;
			// None decoration frame
			if (cmbTemplate.SelectedIndex == 0)
			{
				templatePath += TEMPLATE_SIMPLE;
			}
			// Decoration frame
			else
			{
				templatePath += TEMPLATE_FRAME;
			}

			bpac.DocumentClass doc = new DocumentClass();
			if (doc.Open(templatePath) != false)
			{
				doc.GetObject("objCompany").Text = txtCompany.Text;
				doc.GetObject("objName").Text = txtName.Text;

				// doc.SetMediaById(doc.Printer.GetMediaId(), true);
				doc.StartPrint("", PrintOptionConstants.bpoDefault);
				doc.PrintOut(1, PrintOptionConstants.bpoDefault);
				doc.EndPrint();
				doc.Close();
			}
			else
			{
				MessageBox.Show("Open() Error: " + doc.ErrorCode);
			}
		}

		private void btnExit_Click(object sender, EventArgs e)
		{
			this.Close();
		}
    }
}